eclair-core/src/main/scala/fr/acinq/eclair/io/OpenChannelInterceptor.scala

```diff
@@ -148,6 +148,10 @@ private class OpenChannelInterceptor(peer: ActorRef[Any],
 
   private def sanityCheckNonInitiator(request: OpenChannelNonInitiator): Behavior[Command] = {
     validateRemoteChannelType(request.temporaryChannelId, request.channelFlags, request.channelType_opt, request.localFeatures, request.remoteFeatures) match {
+      case Right(_: ChannelTypes.StaticRemoteKey) if !nodeParams.channelConf.acceptIncomingStaticRemoteKeyChannels =>
```

Comment:
This should also reject legacy channels (channels that don't even have `static_remote_key`), is that also covered? If not, we should:

- always reject channels that don't even have `static_remotekey`
- reject `static_remotekey` channels unless the config flag is set

This may require a bit more work to update tests (to at least use `static_remotekey`), but it does reflect the reality of channels on the network today so it's a good test clean-up to do.

Reply:
done in https://github.com/ACINQ/eclair/pull/2881/commits/e785f5e8d81c7c2e39fee8026bf3db6b63f35a58, which is a fairly large change as most of our unit tests used "standard" (obsolete) channels.